src/lib/ngsi/EntityId.cpp

```diff
@@ -312,6 +306,11 @@ bool EntityId::equal(EntityId* eP)
     return true;
   }
 
+  if (eP->isTypePattern == eP->isTypePattern)
+  {
+    return true;
+  }
+
   return false;
```

Comment:
This I don't like very much ...

Before

    if (eP->isTypePattern == eP->isTypePattern)

there is a 

      if (eP->isPatternIsTrue() == isPatternIsTrue())
      {
        return true;
      }

That was OK, as long as it was *the last if*.
Now it is not, so, if `eP->isPatternIsTrue() != isPatternIsTrue()`, the `true` will not be returned and we continue to `if  eP->isTypePattern == eP->isTypePattern`, which will return `true` in case of equality.
However, isPattern differs and `true` must *not* be returned.
The function must be rewritten like this:

    bool EntityId::equal(EntityId* eP)
    {
      if (eP->id != id)
      {
        return false;
      }

      if (eP->type != type)
      {
        return false;
      }

      if (eP->isPatternIsTrue() != isPatternIsTrue())
      {
        return false;
      }
    
      if (eP->isTypePattern != eP->isTypePattern)
      {
        return false;
      }
    
    ￼  return true;
    }